<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <meta http-equiv="X-UA-Compatible" content="ie=edge">
  <title>08_promise的几个关键问题3</title>
</head>
<body>
  <script>
    /* 
    4.	promise.then()返回的是一个新的Promise实例，它的结果和状态由什么决定?
					(1)简单表达: 由then()所指定的回调函数执行的结果决定
					(2)详细表达:
							①如果抛出异常, 新Promise实例状态为rejected, reason为抛出的异常
							②如果返回的是非Promise的任意值, 新Promise实例状态为resolved, value为返回的值
							③如果返回的是一个Promise实例, 则新Promise实例的状态和结果与返回的那个Promise实例一致
				
				promise.catch()返回的也是新的Promise实例，它的结果和状态由什么决定?
					(1)简单表达: 由catch()所指定的失败的回调函数执行的结果决定
					(2)详细表达:
							①如果抛出异常, 新Promise实例状态为rejected, reason为抛出的异常
							②如果返回的是非Promise的任意值, 新Promise实例状态为resolved, value为返回的值
							③如果返回的是一个Promise实例, 则新Promise实例的状态和结果与返回的那个Promise实例一致
    */

    let p1 =  new Promise((resolve,reject)=>{
      setTimeout(()=>{
        reject(100)
      },1000)
    })

    p1.then(
      (value)=>{console.log('p1成功了',value);return 1},
      (reason)=>{console.log('p1失败了',reason)}
    ).then(
      (value)=>{console.log('x成功了',value)},
      (reason)=>{console.log('x失败了',reason)}
    )



  </script>
</body>
</html>